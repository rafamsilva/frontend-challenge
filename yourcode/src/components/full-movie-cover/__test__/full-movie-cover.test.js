import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import FullMovieCover from '../';

describe('FullMovieCover Component', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  describe('Render', () => {
    it('should render without crashing', () => {
      act(() => {
        render(<FullMovieCover />, container);
      });
    });

    it('should render without image', () => {
      act(() => {
        render(<FullMovieCover />, container);
      });

      expect(container.querySelector("[data-testid='image']")).toBeNull();
    });

    it('should render with image', () => {
      act(() => {
        render(<FullMovieCover image="http://www.fillmurray.com.br/100/100" />, container);
      });

      expect(container.querySelector("[data-testid='image']")).toBeDefined();
    });
  });

  describe('Snapshot', () => {
    it('should match snapshot without image', () => {
      act(() => {
        render(<FullMovieCover />, container);
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });

    it('should match snapshot with image', () => {
      act(() => {
        render(<FullMovieCover image="http://www.fillmurray.com.br/100/100" />, container);
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
  });
});
