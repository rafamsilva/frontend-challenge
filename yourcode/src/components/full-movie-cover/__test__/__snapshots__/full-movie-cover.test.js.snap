// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullMovieCover Component Snapshot should match snapshot with image 1`] = `"<div class=\\"sc-AxjAm lbsqek\\"><img data-testid=\\"image\\" src=\\"http://www.fillmurray.com.br/100/100\\"></div>"`;

exports[`FullMovieCover Component Snapshot should match snapshot without image 1`] = `"<div class=\\"sc-AxjAm lbsqek\\"></div>"`;
